script/zonemaster_backend_testagent

```diff
@@ -87,13 +87,14 @@ sub main {
     my $self = shift;
 
     my $caught_sigterm = 0;
-    local $SIG{TERM} = \&catch_sigterm;
-    sub catch_sigterm {
-        $SIG{TERM} = \&catch_sigterm;
+    my $catch_sigterm;
+    $catch_sigterm = sub {
+        $SIG{TERM} = $catch_sigterm;
```

Comment:
Why is this needed?

Reply:
On some systems the signal handler gets cleared automatically so you have to re-enable it every time. I'm not sure exactly which systems have this behavior so I just threw it in for good measure. In any case it doesn't hurt.